Add function to generate dita image-map from html image-map

ebenezer-melkamu opened the pull request:
Fixes #263 

### Modifications
1. Added a function to copy the `world-map.gif` file from the `data` folder to `regions` folder

ebenezer-melkamu committed:
Add function to copy image from data

ebenezer-melkamu commented:
@IanMayo Should i use libraries like `ElementTree` to create the `PD_1.html` file from my python code, something like this

```py
  
import xml.etree.ElementTree as ET

# create the root element
root = ET.Element("map")
root.set("name", "image-map")

# create the image element
img = ET.Element("image")
img.set("src", "./Content/Images/world-map.gif")
img.set("width", "800")
img.set("height", "600")
root.append(img)

```

IanMayo commented:
> @IanMayo Should i use libraries like `ElementTree` to create the `PD_1.html` file from my python code, something like this 

yes, but you are creating `world-map.dita`.  The `.dita` files have a slightly different format for some elements, so check out the mock version I've linked to in the issue.

ebenezer-melkamu committed:
Generate ditamap from html file

ebenezer-melkamu commented:
@IanMayo, I have successfully generated the `regions.dita` file from the `html` file. Can you please provide me with some guidance for the final step? Do I need to create folders in the `regions` directory based on the `alt` attribute of the country names?

IanMayo commented on target/dita/regions/regions.dita:
```diff
@@ -0,0 +1,48 @@
+    <coords>
+     550,211,522,231,515,263,540,275,564,269,561,231,555,214
+    </coords>
+    <xref format="dita" href="../Britain/Britain1.html"/>
```
this XREF will fail, since there isn't an html file at `../Britain/Britain1.html`.  Instead, we'll just link to the current file, so it's `./regions.dita`.

We'll create these country folders (and files) in the next issue.

IanMayo commented on target/dita/regions/regions.dita:
```diff
@@ -0,0 +1,48 @@
+ </title>
+ <body>
+  <imagemap>
+   <image href="./Content/Images/world-map.gif"/>
```
the parser needs to include the step to copy this source image file into `target/dita/regions/Content/Images`

ebenezer-melkamu replied:
ok, what about this step https://github.com/DeepBlueCLtd/LegacyMan/blob/96336c5408c30edde35010c0a1a69c8c481c94f0/parser/parser.py#L39

IanMayo replied:
Aah, it could be case sensitive.  Let's keep it all lower-case.

IanMayo commented:
> @IanMayo, I have successfully generated the `regions.dita` file from the `html` file. Can you please provide me with some guidance for the final step? Do I need to create folders in the `regions` directory based on the `alt` attribute of the country names?

hopefully the inline feedback answers your questions

ebenezer-melkamu committed:
Change target path for world-map file

IanMayo commented:
Note: I want our process to build up. So, we should still perform the previous steps. 

(the script is currently failing if `target` is missing, but that's just because the early steps aren't being run)

IanMayo commented:
There are some inconsistencies in the generated DITA.  I'll push them in a commit - so you can see in the `delta` what needs to be fixed.

IanMayo committed:
minor fixes:
1. XML definition needs real angle brackets
2. Shape must contain just the element, no newlines.

IanMayo commented on target/dita/regions/regions.dita:
```diff
@@ -0,0 +1,40 @@
+   </area>
+   <area>
+    <shape>poly</shape>
+    <coords>
```
actually @AbegaM - it would be easier to read if the `coords` elements were all on one line, too.

IanMayo commented on target/dita/regions/regions.dita:
```diff
@@ -8,41 +8,33 @@
-    <shape>
-     poly
-    </shape>
+    <shape>poly</shape>
```
aah, @AbegaM - it's the `prettyfy` that is mangling this.
https://github.com/DeepBlueCLtd/LegacyMan/pull/268/files#diff-a6c6b35454c4bcb3844af5bc2c94deb5fd832767ce4eca996361fb50043a4121R105

Let's think about how to resolve it tomorrow.  I _guess_ we could run prettify, then re-process the tree to remove newlines from `<shape>` elements, then write that.  We may end up doing that a lot - so it should be a method that receives the type of element to _Clean-Up_.

ebenezer-melkamu replied:
@IanMayo I have pushed a commit, instead of using `prettify` i have used the `xml.dom.minidom` library to prettify the code, can you run the code and check the `regions.dita`, please.

ebenezer-melkamu committed:
Add function to prettify xml code

IanMayo commented on target/dita/regions/regions.dita:
```diff
@@ -0,0 +1,32 @@
+	<title>Regions</title>
+	<body>
+		<imagemap>
+			<image href="./Content/Images/world-map.gif"/>
```
the above path needs to be lower-case, please @AbegaM 

The rest is fine :-D

ebenezer-melkamu committed:
Convert href to lowercase

IanMayo reviewed (changes requested):
Aah, could you modify the script so it runs the full process?  Some is currently commented out.

ebenezer-melkamu committed:
Uncomment previously commented lines of code

IanMayo committed:
Update root index to include regions

IanMayo commented:
I have an issue.  The `dita` command is failing. Are we running the command before we have produced the `regions.dita`?

```
ian@Ians-Mac-mini LegacyMan % python -m parser.parser data
LegacyMan parser running, with these arguments: ['data']
Target directory does not exist
target directory created
 [gen-list] [DOTX008E][ERROR] File 'file:/Users/ian/git/LegacyMan/target/dita/regions/regions.dita' does not exist or cannot be loaded.
[move-meta] file:/Users/ian/git/LegacyMan/target/dita/index.ditamap:9:59: [DOTX023W][WARN]: Unable to retrieve navtitle from target: 'regions/regions.dita'. 
[move-meta] file:/Users/ian/git/LegacyMan/target/dita/index.ditamap:9:59: [DOTX027W][WARN]: Unable to retrieve linktext from target: 'regions/regions.dita'. 
     [xslt] file:/Users/ian/git/LegacyMan/target/dita/index.ditamap:9:59: [DOTX009W][WARN]: Could not retrieve a title from 'regions/regions.dita'. Using 'regions/regions.dita' instead. 
Publish complete after 2.601580858230591 seconds. Root file at /target/dita/index.ditamap
The directory target/dita/regions already exists
ian@Ians-Mac-mini LegacyMan % ls target/dita/regions 
```

IanMayo committed:
process the regions before we do export

IanMayo committed:
updated export (including regions)

IanMayo commented:
> I have an issue. The `dita` command is failing. Are we running the command before we have produced the `regions.dita`?
> 
> ```
> ian@Ians-Mac-mini LegacyMan % python -m parser.parser data
> ```

It took me way too long to spot that bug.  You won't have seen it, since it's only since I'm adding `regions` to the document:
https://github.com/DeepBlueCLtd/LegacyMan/pull/268/files#diff-115820dc304c8940e5dcc42bcda1e9de59c0ddd3a5751ccf0413442abbdcd850

ebenezer-melkamu commented:
@IanMayo I couldn't reproduce the error in my end

IanMayo commented:
> @IanMayo I couldn't reproduce the error in my end

that's ok. 

in a meeting for the next hour. You ok getting started on the next step?

ebenezer-melkamu commented:
@IanMayo yes i am trying to understand the tasks and the output, i am a little confused but i will figure it out
